cypress/e2e/signIn.cy.js

```diff
@@ -12,23 +17,54 @@ describe('Sign In page', () => {
 
   before(() => {
     cy.task('db:clear');
+  });
+
+  beforeEach(() => {
     cy.task('generateUser').then((generateUser) => {
       user = generateUser;
+      signInPage.visit();
+      cy.registerUser(user.email, user.username, user.password);
     });
   });
 
   it('should provide an ability to log in with existing credentials', () => {
-    signInPage.visit();
-    cy.register(user.email, user.username, user.password);
-
-    signInPage.typeEmail(user.email);
-    signInPage.typePassword(user.password);
-    signInPage.clickSignInBtn();
+    signInPage.enterLoginCredentials(user.email, user.password);
+    signInPage.clickOnSignInButton();
 
     homePage.assertHeaderContainUsername(user.username);
   });
 
-  it('should not provide an ability to log in with wrong credentials', () => {
+  it('should not provide an ability to log in with wrong email', () => {
+    const userEmail = '77' + user.email;
+    signInPage.enterLoginCredentials(userEmail, user.password);
+    signInPage.clickOnSignInButton();
```

Comment:
The selectors used for the alert message and its contents are very general. It is recommended to use `data-qa` attributes for more specific selection.

Reply:
![Uploading e2e_17.png…]()